httpclient/httpclient.go

```diff
@@ -740,3 +753,30 @@ func doneRetrying(err error) error {
 	}
 	return err
 }
+
+func (r Request) validate() (err error) {
```

Comment:
~ ~ The logic of `r.validate` expecting validators to return `false` on valid use cases breaks my brain a bit here. What is the reasoning for going for `false` as the expected happy path return instead of `true` ? My gut would normally say you'd expect a positive for a valid result and a negative for an invalid.